cdap-ui/app/cdap/components/PluginJSONCreator/Create/Content/ConfigurationGroupsCollection/index.tsx

```diff
@@ -126,21 +128,23 @@ const ConfigurationGroupsCollectionView: React.FC<ICreateContext & WithStyles<ty
       setActiveGroupIndex(null);
 
       // Delete a group at the specified index
-      const newGroups = [...localConfigurationGroups];
-      const groupToDelete = newGroups[index];
-      newGroups.splice(index, 1);
+      const groupToDelete = localConfigurationGroups.get(index);
+
+      const newGroups = localConfigurationGroups.remove(index);
       setLocalConfigurationGroups(newGroups);
 
       // Delete the corresponding data of the group
-      const { [groupToDelete]: info, ...restGroupToInfo } = localGroupToInfo;
-      const { [groupToDelete]: widgets, ...restGroupToWidgets } = localGroupToWidgets;
-      setLocalGroupToInfo(restGroupToInfo);
-      setLocalGroupToWidgets(restGroupToWidgets);
+      const newGroupToInfo = fromJS(localGroupToInfo.delete(groupToDelete));
```

Comment:
is localGroupToInfo already an immutable object? If it is, then you don't have to call fromJS. If it's not immutable, why not?

Reply:
Resolved